server/src/main/java/org/opensearch/action/search/SearchRequestContext.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.action.search;
+
+import org.apache.lucene.search.TotalHits;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * This class holds request-level context for search queries at the coordinator node
+ *
+ * @opensearch.internal
+ */
+public class SearchRequestContext {
+    private long absoluteStartNanos;
+    private final Map<String, Long> phaseTookMap;
+    private TotalHits totalHits;
+    private Map<String, Integer> shardStats;
+
+    public SearchRequestContext() {
+        this.phaseTookMap = new HashMap<>();
+        this.totalHits = null;
+    }
+
+    protected void updatePhaseTookMap(String phaseName, Long tookTime) {
+        this.phaseTookMap.put(phaseName, tookTime);
+    }
+
+    protected Map<String, Long> phaseTookMap() {
```

Comment:
Why are these methods `protected`? It doesn't look like this class is designed to be extended. Should they be package-private?

On that note, we should probably make it `final`. [Item 17 in Effective Java](https://catdir.loc.gov/catdir/toc/fy0805/2008926278.html) -- "Design and document for inheritance or else prohibit it".

Reply:
Changed the class to final and methods package private